ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -995,4 +1011,143 @@ void markStartTime() {
       callbackTracker.markOperationStart();
     }
   }
+
+  /**
+   * Injects {@link Account} and {@link Container} for the old put request, which does not carry account or container
+   * header.
+   * @param restRequest The {@link RestRequest} to inject {@link Account} and {@link Container} object.
+   * @throws RestServiceException if either of {@link Account} or {@link Container} object could not be identified.
+   */
+  private void injectAccountAndContainerForOldPutRequest(RestRequest restRequest) throws RestServiceException {
+    Account targetAccount = accountService.getAccountByName(getServiceId(restRequest));
+    if (targetAccount == null) {
+      throw new RestServiceException("Invalid account for putting blob", RestServiceErrorCode.InvalidAccount);
+    }
+    Container targetContainer;
+    if (getIsPrivateSetting(restRequest.getArgs())) {
+      targetContainer = targetAccount.getContainerById(accountService.getContainerIdForLegacyPutPrivateBlob());
+    } else {
+      targetContainer = targetAccount.getContainerById(accountService.getContainerIdForLegacyPutPublicBlob());
+    }
+    if (targetContainer == null) {
```

Comment:
can targetContainer be null incase of targetAccount being non null for old put request?